### PR TITLE
IDE

### PR DESCRIPTION
**Pull request by levkk**
- Added an IDE to execute SQL queries in the dashboard.
- Fixed `.python-version` to work with pyenv.
- A bit of linting.

image

### CODE REVIEW

File: pgml-dashboard/app/templates/base.html [hunk 1]
```diff
@@ -21,6 +21,12 @@
         <script defer type="text/javascript" src="{% static '/js/vendor/lodash-4.17.21.min.js' %}"></script>
         <script defer type="module" src="https://unpkg.com/stimulus@2.0.0/dist/stimulus.umd.js"></script>
         <script defer type="module" src="https://cdn.skypack.dev/pin/@hotwired/turbo@v7.0.0-rc.1-Vumm1JJjKMC0HM7D9Tjd/mode=imports/optimized/@hotwired/turbo.js"></script>
+        <script defer type="module" src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/6.65.7/codemirror.min.js"></script>
```
Comment (montanalow) [inline pgml-dashboard/app/templates/base.html:24]: How are these generated with the integrity values?

File: pgml-dashboard/app/templates/projects/sample.html [hunk 2]
```diff
@@ -15,10 +15,12 @@
         </tr>
         {% endfor %}
 
+        {% if redacted %}
```
Comment (montanalow) [inline pgml-dashboard/app/templates/projects/sample.html:18]: top secret

File: pgml-dashboard/app/views/ide.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    "columns": [desc[0] for desc in cursor.description],
+                    "rows": results,
+                },
+            )
+        except Exception as e:
+            return HttpResponse(
+                f"""
+                <code>
+<pre>
```
Comment (montanalow) [inline pgml-dashboard/app/views/ide.py:63]: formatting?

**Review by montanalow — COMMENTED** [review]
🔥

File: pgml-dashboard/app/templates/base.html (hunk 1 above)
Comment (levkk) [reply pgml-dashboard/app/templates/base.html:24]: Just a link from that site that indexes all the CDNs for js/css libs.

File: pgml-dashboard/app/templates/projects/sample.html (hunk 2 above)
Comment (levkk) [reply pgml-dashboard/app/templates/projects/sample.html:18]: Haha yeah that's not the right word :)

File: pgml-dashboard/app/views/ide.py (hunk 3 above)
Comment (levkk) [reply pgml-dashboard/app/views/ide.py:63]: `pre` preserves formatting, incl indentation, so this was my quick way of aligning it on the page.

**Comment by montanalow** [comment]
Why isn’t our code tag pre as well via css to remove the redundancy?

**Comment by levkk** [comment]
> 

Good question...because `code` doesn't preserve indentation? Fun fact, textarea preserves indent too.

**Pull request merged**